src/ConnectionDispatcher.hpp

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include <memory>
+
+#include "DBConnection2.hpp"
+
+class IConnectionDispatcher {
+public:
+    virtual void connect(MySqlDBConnection2& connection) = 0;
+    virtual void connect(SqLiteDBConnection2& connection) = 0;
+};
+
+class MySqlConnectionDispatcher : public IConnectionDispatcher {
+public:
+    MySqlConnectionDispatcher() : m_mysql_connection(nullptr) {};
+
+    void connect(MySqlDBConnection2& connection) override;
```

Comment:
@morheit, i've asked before to rename connect to dispatch. I see connect. Why?

Reply:
@malirod , I remember that you've asked for this in class `MySqlDBConnection2`. In `MySqlDBConnection2` it's implemented. Do we need the same method's names in `MySqlDBConnection2` and `ConnectionDispatcher`?

Reply:
@malirod , `connect` replaced by `dispatch` in `ConnectionDispatcher`